File: myTargetDemo/app/src/main/java/com/my/targetDemoApp/fragments/PlusDialogFragment.java
package com.my.targetDemoApp.fragments;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;
import android.view.View;
import android.view.WindowManager;
import android.widget.EditText;
import android.widget.FrameLayout;
import android.widget.RadioGroup;
import android.widget.Toast;

import com.my.targetDemoApp.AdTypes;
import com.my.targetDemoApp.R;

public class PlusDialogFragment extends DialogFragment
{

	private SaveTypeListener saveTypeListener;
	private EditText editText;
	private RadioGroup radioGroup;

	@Override
	public void onActivityCreated(Bundle savedInstanceState)
	{
		super.onActivityCreated(savedInstanceState);
		getDialog().getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_VISIBLE);
	}

	@NonNull
	@Override
	public Dialog onCreateDialog(Bundle savedInstanceState)
	{
		FrameLayout dialogLayout = new FrameLayout(getContext());
		View v = getActivity().getLayoutInflater().inflate(R.layout.fragment_dialog, dialogLayout,
				false);

		editText = (EditText) v.findViewById(R.id.editText);
		radioGroup = (RadioGroup) v.findViewById(R.id.radiogroup);

		dialogLayout.addView(v);

		return new AlertDialog.Builder(getActivity())
				.setView(dialogLayout)
				.setPositiveButton(R.string.ok, okListener)
				.setNegativeButton(R.string.cancel, cancelListener)
				.setTitle(R.string.add_ad_title)
				.create();
	}

	private DialogInterface.OnClickListener okListener = new DialogInterface.OnClickListener()
	{
		@Override
		public void onClick(DialogInterface dialog, int which)
		{
			if (saveTypeListener != null)
			{
				try
				{
					int slotId = Integer.parseInt(editText.getText().toString());
					int checkedAdType;
					switch (radioGroup.getCheckedRadioButtonId())
					{
						case R.id.adtype_banner_320x50:
						default:
							checkedAdType = AdTypes.AD_TYPE_320X50;
							break;
						case R.id.adtype_banner_300x250:
							checkedAdType = AdTypes.AD_TYPE_300X250;
							break;
						case R.id.adtype_banner_728x90:
							checkedAdType = AdTypes.AD_TYPE_728X90;
							break;
						case R.id.adtype_interstitial:
							checkedAdType = AdTypes.AD_TYPE_FULLSCREEN;
							break;
						case R.id.adtype_instream:
							checkedAdType = AdTypes.AD_TYPE_INSTREAM;
							break;
						case R.id.adtype_native:
							checkedAdType = AdTypes.AD_TYPE_NATIVE;
							break;
					}

					saveTypeListener.onSaveType(checkedAdType, slotId);
				} catch (NumberFormatException e)
				{
					Toast.makeText(getActivity(), getString(R.string.wrong_slot), Toast.LENGTH_SHORT).show();
					e.printStackTrace();
				}
			}
		}
	};

	private DialogInterface.OnClickListener cancelListener = new DialogInterface.OnClickListener()
	{
		@Override
		public void onClick(DialogInterface dialog, int which)
		{
			dismiss();
		}
	};

	public void setSaveTypeListener(SaveTypeListener saveTypeListener)
	{
		this.saveTypeListener = saveTypeListener;
	}

	public interface SaveTypeListener
	{
		void onSaveType(int adType, int slotId);
	}
}

File: myTargetDemo/app/src/main/java/com/my/targetDemoApp/PlusDialogFragment.kt
package com.my.targetDemoApp

import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import android.text.TextUtils
import android.view.WindowManager
import android.widget.FrameLayout
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import com.my.targetDemoApp.databinding.FragmentDialogBinding

class PlusDialogFragment(val onSave: (CustomAdvertisingType) -> Unit) : DialogFragment() {
    private var binding: FragmentDialogBinding? = null
    private val cancelListener = DialogInterface.OnClickListener { _, _ -> dismiss() }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        dialog?.window?.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_VISIBLE)
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val dialogLayout = FrameLayout(requireContext())
        val binding = FragmentDialogBinding.inflate(layoutInflater, dialogLayout, false)
        this.binding = binding
        binding.btnOk.setOnClickListener {
            val slotId = binding.editText.text?.toString()
                    ?.toIntOrNull()
            if (slotId == null) {
                Toast.makeText(requireContext(), "Null slot, cannot save", Toast.LENGTH_SHORT)
                        .show()
                return@setOnClickListener
            }
            val paramsText: String? = binding.etParams.text?.toString()
            if (!paramsText.isNullOrEmpty()) {
                if (!validateParams(paramsText)) {
                    Toast.makeText(requireContext(),
                            """Wrong custom params, leave blank or format properly
                        """.trimMargin(), Toast.LENGTH_SHORT)
                            .show()
                    return@setOnClickListener
                }
            }

            val type: CustomAdvertisingType.AdType =
                    when (binding.rgChooseAd.checkedRadioButtonId) {
                        R.id.adtype_banner_320x50         -> CustomAdvertisingType.AdType.STANDARD_320X50
                        R.id.adtype_banner_300x250        -> CustomAdvertisingType.AdType.STANDARD_300X250
                        R.id.adtype_banner_728x90         -> CustomAdvertisingType.AdType.STANDARD_728X90
                        R.id.adtype_banner_adaptive       -> CustomAdvertisingType.AdType.STANDARD_ADAPTIVE
                        R.id.adtype_interstitial          -> CustomAdvertisingType.AdType.INTERSTITIAL
                        R.id.adtype_rewarded              -> CustomAdvertisingType.AdType.REWARDED
                        R.id.adtype_instream              -> CustomAdvertisingType.AdType.INSTREAM
                        R.id.adtype_native                -> CustomAdvertisingType.AdType.NATIVE_AD
                        R.id.adtype_native_configurations -> CustomAdvertisingType.AdType.NATIVE_AD_CONFIGURATIONS
                        R.id.adtype_native_banner         -> CustomAdvertisingType.AdType.NATIVE_BANNER
                        R.id.adtype_audio_instream        -> CustomAdvertisingType.AdType.AUDIO_INSTREAM
                        else                              -> CustomAdvertisingType.AdType.STANDARD_320X50
                    }
            val checkedAdType = CustomAdvertisingType(type, slotId, paramsText)
            val name = binding.etName.text.toString()
            if (name.isNotEmpty()) {
                checkedAdType.name = name
            }
            onSave.invoke(checkedAdType)
            dismiss()
        }
        binding.btnCancel.setOnClickListener { dismiss() }

        dialogLayout.addView(binding.root)

        return AlertDialog.Builder(requireActivity())
                .setView(dialogLayout)
                .setTitle(R.string.add_ad_title)
                .create()
    }

    private fun validateParams(paramsText: String): Boolean {
        return try {
            paramsText.split("&")
                    .forEach {
                        val split = it.split("=")
                        if (split.size != 2) return false
                        if (TextUtils.isEmpty(split[0])) {
                            return false
                        }
                    }
            true
        }
        catch (e: Exception) {
            false
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }
}
